nncf/experimental/openvino_native/graph/model_transformer.py

```diff
@@ -125,30 +125,29 @@ def _get_extra_model_outputs(self,
             port_id = transformation.target_point.port_id
             if transformation.target_point.type == TargetType.POST_LAYER_OPERATION:
                 output = node.output(port_id)
-                extra_model_outputs.append(output)
+                extra_model_outputs.append((output, port_id))
             elif transformation.target_point.type == TargetType.PRE_LAYER_OPERATION:
                 output = node.input_value(port_id)
-                extra_model_outputs.append(output)
+                extra_model_outputs.append((output, port_id))
             else:
                 raise NotImplementedError(f'Unsupported target point type {transformation.target_point.type}')
 
         return extra_model_outputs
 
     @staticmethod
-    def _insert_outputs(model: ov.Model, outputs: List[ov.Output]) -> ov.Model:
+    def _insert_outputs(model: ov.Model, outputs: List[Tuple[ov.Output, int]]) -> ov.Model:
         """
         Takes a model and adds outputs based on the list of ov.Output.
 
         :param model: OpenVINO model.
-        :param outputs: list of ov.Output.
+        :param outputs: list of tuples with ov.Output & port_ids.
         :return: modified model.
         """
         model_outputs = model.get_results()
         params = model.get_parameters()
         extra_model_outputs = []
-        for output in outputs:
+        for (output, port_id) in outputs:
             output_name = output.get_node().get_friendly_name()
-            port_id = output.get_index()
             result = opset.result(output, name=f'Result_{output_name}.{port_id}')
```

Comment:
I think we should support nodes with several outputs or add TODO here and open a ticket.

Reply:
Ticket - 103787